erdos/local/local_dispatcher.py

```diff
@@ -0,0 +1,33 @@
+import pexpect
+import time
+
+from erdos.cluster.dispatcher import Dispatcher
+
+class LocalDispatcher(Dispatcher):
+    """Run commands locally"""
+    def __init__(self):
+        super(LocalDispatcher, self).__init__()
+        self.current_process = None
+        self.processes = []
+
+    def run(self, command):
+        self.current_process = pexpect.spawn(command)
+        self.processes.append(self.current_process)
+
+    def wait_for_prompt(self, timeout=10):
+        if self.current_process is None:
+            return b""
+
+        start = time.time()
+        end = start + timeout
+        while time.time() < end:
+            if not self.current_process.isalive():
+                process = self.current_process
+                self.current_process = None
+                self._cleanup_processes()
+                return process.read()
+
```

Comment:
Do you mean if the timeout is violated? Because this checks if the process completes, and if it does, prints all the output which is similar to the command line behavior.

Reply:
Yes, if the timeout is violated (i.e., the process doesn't complete in time).

I just noticed that this busy loop doesn't have a time.sleep. Should we do a time.sleep(smallish_value) so that we give the CPU core some time to do other work? Otherwise, this while loop will use an entire core.